katdal/datasources.py

```diff
@@ -283,16 +283,30 @@ def from_url(cls, url, chunk_store=None):
                 telstate.load_from_file(url_parts.path)
             except OSError as err:
                 raise DataSourceNotFound(str(err))
-            return cls(view_capture_stream(telstate, **kwargs), chunk_store,
-                       source_name)
+            telstate = view_capture_stream(telstate, **kwargs)
+            # Look for adjacent data directory (presumably containing NPY files)
+            if chunk_store == 'auto':
+                capture_block_path = os.path.dirname(url_parts.path)
+                store_path = os.path.join(capture_block_path, '..')
+                try:
+                    data_path = os.path.join(store_path, telstate['chunk_name'])
+                except KeyError:
+                    chunk_store = None
+            if chunk_store == 'auto' and os.path.isdir(data_path):
+                chunk_store = NpyFileChunkStore(os.path.normpath(store_path))
```

Comment:
If you're going to use normpath, do it before checking `isdir`: normpath can alter the meaning of a path when symbolic links are involved. I think it may be a bad idea to use it though, precisely because it may end up being a different path to the one the user asked for.

Reply:
Good point - I'm not really familiar with `normpath` and just thought it was cuter to do go

`bucket/cb` -> `bucket/cb/../cb_stream` -> `bucket/cb_stream`

instead of

`bucket/cb` -> `bucket/` -> `bucket/cb_stream`

If the `..` doesn't hurt I'll leave it in.